### PR TITLE
task: added documentation and images for the secrets feature

### PR DESCRIPTION
**Pull request by swampie**
Added entries inside the Core Concepts + general usage info of both Workspace Secrets and User Secrets

### CODE REVIEW

File: docs/core-concepts/definitions.md [hunk 1]
```diff
@@ -39,6 +39,17 @@ A Compute Environment is composed of credentials, configuration settings and sto
 Credentials are access keys stored by Tower in an encrypted manner using AES-256. They allow safe storage of authentication keys for Compute Environments, private code repositories, and external services. Credentials cannot be accessed once stored.
 
 
+## Secrets
+
+Secrets, as Credentials, are keys that may be used by the pipelines in order to interact with external systems (e.g a password to connect to an external database). These keys are stored inside towe in an encrypted manner using AES-256. Currently there are two main type of secrets:
```
Comment (drpatelh) [inline docs/core-concepts/definitions.md:44]:
```suggestion
Secrets, like Credentials, are keys that may be used by pipelines in order to interact with external systems e.g. a password to connect to an external database. Secrets are stored within Tower using AES-256 encryption. Currently, there are two main types of secrets:
```

**Comment by swampie** [comment]
@abhi18av this documentation should appear only for latest version and not for prior

File: docs/core-concepts/definitions.md [hunk 2]
```diff
@@ -39,6 +39,17 @@ A Compute Environment is composed of credentials, configuration settings and sto
...
+
+- Workspace Secrets are defined at workspace level and will be injected inside the workflow of every pipeline that is defined inside the workspace
```
Comment (drpatelh) [inline docs/core-concepts/definitions.md:46]:
```suggestion
- Workspace-level Secrets are injected into all Pipelines that are launched within a Workspace.
```

File: docs/core-concepts/definitions.md [hunk 3]
```diff
@@ -39,6 +39,17 @@ A Compute Environment is composed of credentials, configuration settings and sto
...
+
+- User Secrets are injected in all the pipeline launched by a specific user
```
Comment (drpatelh) [inline docs/core-concepts/definitions.md:48]:
```suggestion
- User-level Secrets are injected into all Pipelines that are launched by a specific user. 
```

File: docs/core-concepts/definitions.md [hunk 4]
```diff
@@ -39,6 +39,17 @@ A Compute Environment is composed of credentials, configuration settings and sto
...
 
+## Secrets
+
+Secrets, as Credentials, are keys that may be used by the pipelines in order to interact with external systems (e.g a password to connect to an external database). These keys are stored inside towe in an encrypted manner using AES-256. Currently there are two main type of secrets:
+
+- Workspace Secrets are defined at workspace level and will be injected inside the workflow of every pipeline that is defined inside the workspace
+
+- User Secrets are injected in all the pipeline launched by a specific user  
+
+> In case of a name clash the User Secrets have higher priority and will override any Workspace Secret with the same name
```
Comment (drpatelh) [inline docs/core-concepts/definitions.md:50]:
```suggestion
> User-level Secrets have higher priority and will override any Workspace-level Secrets with the same name.
```

File: docs/secrets/overview.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+title: Secrets Overview
+headline: 'Secrets'
+description: 'Step-by-step instructions to set up user and workspace secrets in Nextflow Tower.'
```
Comment (drpatelh) [inline docs/secrets/overview.md:4]:
```suggestion
description: 'Step-by-step instructions to set-up User and Workspace Secrets in Nextflow Tower.'
```

File: docs/secrets/overview.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+## Introduction
+
+Tower uses the concept of **Secrets** to store the keys and tokens that are not meant to be exposed or to be transferred between the platform and the compute environment. In order to achieve this extra level of security, Tower relies on Secrets Manager technology that Cloud providers offers in order to maintain the traffic between the Compute Environment and the vault secure.
```
Comment (drpatelh) [inline docs/secrets/overview.md:9]:
```suggestion
Tower uses the concept of **Secrets** to store private keys and tokens that are not meant to be exposed or transferred between a given computing platform and the Tower Compute Environment. In order to achieve this extra level of security, Tower relies on existing Secrets Manager technologies offered by Cloud providers to maintain the traffic between the Compute Environment and the secure vault.
```

Comment (drpatelh) [reply docs/secrets/overview.md:9]: Which existing technologies are we using here and what does "vault secure" mean? I got a little confused by this sentence:
"In order to achieve this extra level of security, Tower relies on Secrets Manager technology that Cloud providers offers in order to maintain the traffic between the Compute Environment and the vault secure."

File: docs/secrets/overview.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+!!! note 
+    Currently only Slurm/HPC and AWS are supported. Please read more about AWS Secret Manager [here](https://docs.aws.amazon.com/secretsmanager/index.html)
```
Comment (drpatelh) [inline docs/secrets/overview.md:12]:
```suggestion
    Currently, only Slurm/HPC and AWS Batch are supported. You can read more about AWS Secrets Manager [here](https://docs.aws.amazon.com/secretsmanager/index.html)
```

File: docs/secrets/overview.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+## Workspace Secrets
+
+To create a Workspace Secret is necessary to navigate to a workspace first (private or shared). From there a new tab **Secret** in the top navigation will give access to the secrets management interface.
```
Comment (drpatelh) [inline docs/secrets/overview.md:16]:
```suggestion
To create a Workspace-level Secret navigate to a Workspace (private or shared) and click on the **Secrets** tab in the top navigation pane to gain access to the Secrets management interface.
```

File: docs/secrets/overview.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+![](_images/workspace_secrets_and_credentials.png)
+
+The UI shows the list of available secrets and gives allowed users (admin or owner of the workspace) the possibility to create a new secret or update an existing secret value
```
Comment (drpatelh) [inline docs/secrets/overview.md:20]:
```suggestion
All of the available Secrets will be listed here and users with the appropriate permissions (Workspace admin or owner) will be able to create or update Secret values.
```

File: docs/secrets/overview.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+![](_images/secrets_list.png)
+
+The form for creating or edit a secret is very similar to the one used for credentials.
```
Comment (drpatelh) [inline docs/secrets/overview.md:24]:
```suggestion
The form for creating or updating a Secret is very similar to the one used for Credentials.
```

File: docs/secrets/overview.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+![](_images/secrets_creation_form.png)
+
+## User Secrets
+
+The User Secrets list and edit/create form works in the same way as the one described above. The menu that appears clicking the avatar in the top right corner gives access to the functionality
```
Comment (drpatelh) [inline docs/secrets/overview.md:30]:
```suggestion
User-level Secrets can be accessed by clicking on your avatar in the top right corner of the Tower interface and selecting "Your Secrets". Conceptually, listing, creating and updating User-level Secrets is the same as highlighted for Workspace-level Secrets. 
```

File: docs/secrets/overview.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+![](_images/personal_secrets_and_and_credentials.png)
+
+## Secrets Usage in Pipeline Runs
+
+When a new pipeline is launched then all the secrets (workspace and user) are sent to the corresponding compute environment secret manager: nextflow execution will then download these secrets internally and uses these when referenced in the pipeline code.
```
Comment (drpatelh) [inline docs/secrets/overview.md:36]:
```suggestion
When a new Pipeline is launched then all Secrets (Workspace and User) are sent to the corresponding Secrets Manager for the Compute Environment: On executions, Nextflow will download these Secrets internally and uses them wherever they are referenced in the pipeline code.
```

**Review by drpatelh — COMMENTED** [review]
LGTM! Minor changes. One more sentence to review after lunch.

File: docs/core-concepts/definitions.md [hunk 13]
```diff
@@ -39,6 +39,17 @@ A Compute Environment is composed of credentials, configuration settings and sto
...
 
+## Secrets
+
+Secrets, like Credentials, are keys that may be used by pipelines in order to interact with external systems e.g. a password to connect to an external database. Secrets are stored within Tower using AES-256 encryption. Currently, there are two main types of secrets:
+
+- Workspace-level Secrets are injected into all Pipelines that are launched within a Workspace.
+
+- User-level Secrets are injected into all Pipelines that are launched by a specific user. 
+
+> User-level Secrets have higher priority and will override any Workspace-level Secrets with the same name.
```
Comment (abhi18av) [inline docs/core-concepts/definitions.md:50]:
```suggestion

!!! warn
    User-level Secrets have higher priority and will override any Workspace-level Secrets with the same name.
```

Comment (abhi18av) [reply docs/core-concepts/definitions.md:50]: I'm guessing you want this info to stand out?

File: docs/secrets/overview.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
+---
+title: Secrets Overview
+headline: 'Secrets'
+description: 'Step-by-step instructions to set-up User and Workspace Secrets in Nextflow Tower.'
```
Comment (abhi18av) [inline docs/secrets/overview.md:4]: Branding was changed to `Tower` only.

```suggestion
description: 'Step-by-step instructions to set-up User and Workspace Secrets in Tower.'
```

File: docs/secrets/overview.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+## Introduction
+
+Tower uses the concept of **Secrets** to store the keys and tokens that are not meant to be exposed or to be transferred between the platform and the compute environment. In order to achieve this extra level of security, Tower relies on Secrets Manager technology that Cloud providers offers in order to maintain the traffic between the Compute Environment and the vault secure.
```
Comment (abhi18av) [inline docs/secrets/overview.md:9]:
```suggestion
Tower uses the concept of **Secrets** to store the keys and tokens that are not meant to be exposed or to be transferred between different platforms and the compute-environments. In order to achieve this extra level of security, Tower relies on Secrets Manager technology natively offered by Cloud providers to maintain secure traffic between the Compute Environment and the secrets vault.
```

**Review by abhi18av — COMMENTED** [review]
Added minor comments, I'll update the screenshots a bit once this is merged into the staging branch and is visible on the [staging website](https://nf-tower-docs-staging-html.netlify.app/21.12/).

Furthermore, there are a cople points

1. Regarding the placement of the `Secrets` page - is it necessary to create nesting for this 🤔. Perhaps a top-level entry like that for [CLI](https://help.tower.nf/21.12/cli/) would suffice - WDYT? 

Nesting in this context means that `Secrets/` holds pages other than `Overview`. 

2. > this documentation should appear only for latest version and not for prior

Currently, the latest version on tower.nf is `Version 22.1.0-dalambert_feaff2c6` and on the docs website is synced with the **latest enterprise** release - shall I bump the docs website version to `v22.1` moving forward?

File: docs/secrets/overview.md [hunk 16]
```diff
@@ -35,4 +35,8 @@ User-level Secrets can be accessed by clicking on your avatar in the top right c
 
 When a new Pipeline is launched then all Secrets (Workspace and User) are sent to the corresponding Secrets Manager for the Compute Environment: On executions, Nextflow will download these Secrets internally and uses them wherever they are referenced in the pipeline code.
 
-Secrets will be automatically deleted from the Secrets Manager when the Pipeline reaches completion (successful or unsuccessful).
\ No newline at end of file
+Secrets will be automatically deleted from the Secrets Manager when the Pipeline reaches completion (successful or unsuccessful).
+
+## Aws Secrets Manager Integration
```
Comment (abhi18av) [inline docs/secrets/overview.md:40]:
```suggestion
## AWS Secrets Manager Integration
```

**Comment by abhi18av** [comment]
@swampie , please confirm if the remaining conversations are valid - if so, please resolve. I think the PR is good to merge and deploy to staging website.


Furthermore, I'd appreciate your thoughts on the questions mentioned here https://github.com/seqeralabs/nf-tower-docs/pull/125#pullrequestreview-897546912

**Comment by swampie** [comment]
@abhi18av Confirmed with Paolo that next release 22.1.X will be the container of all the new upcoming features including secrets.
I need your help to structure the documentation in order to hide secrets from previous versions

File: docs/secrets/overview.md (hunk 15 above)
Comment (swampie) [reply docs/secrets/overview.md:9]: It's not different platforms but Tower

**Comment by abhi18av** [comment]
Update - this page currently looks like this https://nf-tower-docs-staging-html.netlify.app/22.1/secrets/overview/ 

I understand that there's a Google doc and there might be further updates, but having a preview of current status on the website might inform the new content better.

**Pull request merged**